symphony-bdk-http/symphony-bdk-http-webclient/src/main/java/com/symphony/bdk/http/webclient/ApiClientWebClient.java

```diff
@@ -0,0 +1,287 @@
+            File file = (File) param.getValue();
+            formValueMap.add(param.getKey(), file);
+          } else {
+            formValueMap.add(param.getKey(), parameterToString(param.getValue()));
+          }
+        }
+        requestBodySpec.body(BodyInserters.fromMultipartData(formValueMap));
+      } else if (contentType.startsWith("application/x-www-form-urlencoded")) {
+        MultiValueMap<String, String> formValueMap = new LinkedMultiValueMap<>();
+        for (Map.Entry<String, Object> param : formParams.entrySet()) {
+          formValueMap.add(param.getKey(), parameterToString(param.getValue()));
+        }
+        requestBodySpec.body(BodyInserters.fromFormData(formValueMap));
+      }
+    }
+    if (body != null) {
+      requestBodySpec.body(BodyInserters.fromValue(body));
+    }
+
+    Mono<ClientResponse> mono = requestBodySpec.exchange();
+    ClientResponse response = mono.block();
+    Map<String, List<String>> headers = new HashMap<>();
+    for (Map.Entry<String, List<String>> entry : response.headers().asHttpHeaders().entrySet()) {
+      headers.put(entry.getKey(), entry.getValue());
+    }
+
+    if (response.statusCode().equals(HttpStatus.NO_CONTENT)) {
+      return new ApiResponse<>(response.statusCode().value(), headers);
+    } else if (response.statusCode().equals(HttpStatus.OK)) {
+      if (returnType == null) {
+        return new ApiResponse<>(response.statusCode().value(), headers);
+      } else {
+        if (returnType.getType() instanceof Class) {
+          Class<T> clazz = (Class<T>) returnType.getType();
+          T entity = response.bodyToMono(clazz).block();
+          return new ApiResponse<>(response.statusCode().value(), headers, entity);
+        } else {
+          ParameterizedTypeReference<T> reference = ParameterizedTypeReference.forType(returnType.getType());
+          T entity = response.bodyToMono(reference).block();
+          return new ApiResponse<>(response.statusCode().value(), headers, entity);
+        }
+      }
+    } else {
+      String message = "error";
+      String respBody = response.bodyToMono(String.class).block();
+      if (respBody != null) {
+        message = respBody;
+      }
+
+      throw new ApiException(
+          response.rawStatusCode(),
+          message,
+          headers,
+          respBody);
+    }
+  }
+
+  @Override
+  public String getBasePath() {
+    return this.basePath;
+  }
+
+  @Override
+  public String parameterToString(Object param) {
+    if (param == null) {
+      return "";
+    } else if (param instanceof Collection) {
+      StringBuilder b = new StringBuilder();
+      for (Object o : (Collection<?>) param) {
+        if (b.length() > 0) {
+          b.append(',');
+        }
+        b.append(o);
+      }
+      return b.toString();
+    } else {
+      return String.valueOf(param);
+    }
+  }
+
+  @Override
+  public List<Pair> parameterToPairs(String collectionFormat, String name, Object value) {
```

Comment:
I don't understand this method (and how it is going to be used), could you give a bit of context?

Reply:
It is unfortunately used by generated code